jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainActivity.java

```diff
@@ -35,11 +45,23 @@
     // Maps with layout values.
     private Map<String, LinearLayout.LayoutParams> fragmentLayouts = new HashMap<>();
 
+    // Files
+    private File messageFile = null;
+
+    // Filenames
+    private final String messageFilename = "messageFile";
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
         setupFragments();
+
+        try {
+            messageFile = File.createTempFile(messageFilename, null, this.getCacheDir());
```

Comment:
Why are you saving persistent data as a cache file in the cache directory? What happens if the user deletes the cache folder?